runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -1140,6 +1140,7 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
                     } catch (WrongClusterException we) {
                         // It is futile trying to re-connect to the wrong cluster
                         log.warn("Giving up since cluster is incorrect or reconfigured!");
+                        parameters.getSystemDownHandler().run();
                         throw we;
```

Comment:
Should the exception still be thrown after the handler is called ?

Reply:
the reason for throwing the exception is some shutdown handlers may just be no-ops (by intent), in which case we want to still fail the operation.